plugins/ICEventInfoProducer.cc

```diff
@@ -158,13 +159,20 @@ void ICEventInfoProducer::produce(edm::Event& event,
   }
 
   edm::Handle<LHEEventProduct> lhe_handle;
+  edm::Handle<GenEventInfoProduct> gen_info_handle;
+  event.getByLabel("generator",gen_info_handle);
+  if(gen_info_handle->weight()>=0){
```

Comment:
won't this crash on data? Plus I could still imagine there are MC samples out there that don't have the "generator" object so I think it would be good not to require this to be present in the event